### PR TITLE
Breaking change: Replacing `match_type` with `mapping_justification`

### PR DESCRIPTION
**Pull request by matentzn**
See also #150

After some deliberations with some of you, this will be the hopefully last major breaking change in SSSOM.

- [ ] Note to self, make comment on https://github.com/agroportal/project-management/issues/265 after PR is merged

### CODE REVIEW

File: model/schema/sssom.yaml [hunk 1]
```diff
@@ -162,15 +154,14 @@ slots:
...
-    description: The kind of match that led to the mapping, e.g. Logical or Lexical.
-    range: match_type_enum
+  mapping_justification:
+    description: A mapping justification is an action (or the written representation of that action) of showing a mapping to be right or reasonable.
+    range: EntityReference
     required: true
-    multivalued: true
     examples:
-      - value: Lexical
+      - value: SSSOMV:Lexical
```
Comment (cthoyt) [inline model/schema/sssom.yaml:162]: Before evaluating if this is a good change for SSSOM, I think the `SSSOMV` vocabulary needs to get fleshed out a bit more? I.e., is there finally a page that is specifically describing all of the terms? Previously, there were some explanations of a few of the possible terms buried in a pile of other irrelevant documentation, which isn't good enough.

This vocabulary also needs a Bioregistry entry ;)

Comment (matentzn) [reply model/schema/sssom.yaml:162]: Maybe I should remove the example - This vocabulary does not exist yet.. Not even sure its should have the SSSOM namespace, or another: Something like "Vocabulary for Ontology Mapping"? That could be a cool standalone thing to do anyways, and then the prefix wouldn't be SSSOMV. What do you think? A vocabulary in the SSSOM namespace or adjacent (SSSOMV) or something independent in mapping commons org: Vocabulary for Ontology Matching?

Comment (graybeal) [reply model/schema/sssom.yaml:162]: I too am having trouble evaluating this change. Partly understanding/tracking the purpose of the diverse changes, partly understanding their applications (what does it mean when permission_values are or are not listed?), partly interpreting the `value:` string generally (what does it mean that most do not have full IRIs or CURIEs? are they not in the vocabulary)?  Some of the definitions, n.b. Logical and Lexical, are inadequate, and some of the choices are overlapping.

My observation on that particular point is that the vocabulary name/IRI should be defined as part of the release. Include SSSOMV in the prefix list and make it `https://w3id.org/sssom/schema/voc/` (please note the https protocol, which SSSOM should also be define as). You already have the w3id prefix so this should not be an obstacle to implementation at any point.

**Comment by cthoyt** [comment]
@matentzn btw it's already sort of a problem that the O in SSSOM stands for ontologies since there are so many other kinds of semantic space formats. if you want to mint a new vocabulary to describe mappings, I'd suggest not making it specifically named for ontologies

File: model/schema/sssom.yaml (hunk 1 above)
Comment (matentzn) [reply model/schema/sssom.yaml]: For the sake of explaining the situation, I created the vocabulary I envision here, with only 3 classes: https://github.com/mapping-commons/ontological-mapping-vocabulary. I propose to build a separate vocabulary, independent of SSSOM, to capture all the activities, match types etc as they are currently prevalent in the mapping community. I know about all the disagreements on terminology, so let us not make that an issue here - this is already in SSSOM - so not currently up for debate.

So this PR should not be about evaluating that vocabulary - I will complete it with what we anyways have right now in SSSOM once the general approach is signed off on. And then anyone can add new terms in there.

The only thing that matters here is this:

- We change the `match_type` field to `mapping_justification` for the reasons outlined in #150 
- We constrain the `mapping_justification` field to be values from a controlled semantic web vocabulary.

**Comment by matentzn** [comment]
@cthoyt Name suggestions welcome :P

**Review by cmungall — APPROVED** [review]

**Comment by matentzn** [comment]
@callahantiff @graybeal can you also take another look?

**Review by graybeal — APPROVED** [review]

**Comment by matentzn** [comment]
Thank you all for the review!

**Comment by callahantiff** [comment]
So sorry I was unable to review in time @matentzn. I will make sure to respond quicker in the future!

**Comment by matentzn** [comment]
NP I would have waited, am just in a rush on some features :)

**Comment by callahantiff** [comment]
> NP I would have waited, am just in a rush on some features :)

I completely understand! 😄

**Pull request merged**